src/app/users/users.component.ts

```diff
@@ -102,12 +101,15 @@ export class UsersComponent implements OnInit, AfterViewInit {
     });
   }
 
-  deleteRole(user: any, index: number) {
+  deleteRole(user: any, index: number, position: number) {
     // Make copy of user so UI doesn't change until DB change succeeds
-    const tempUser = { ...user, roles: [ ...user.roles ] };
+    let tempUser = { ...user, roles: [ ...user.roles ] };

```

Comment:
i'm not getting code as it seems to update tempUser roles from user.roles which should have already been there when you make tempUser equals to user

Reply:
@lmmrssa This is a key part of JavaScript: deep vs. shallow copy. `...` is one way for doing a shallow copy of an object/array.  In this situation since the `roles` property is also an object/array we need to also use `...` on that object/array to create a copy.